### DISCUSSION TITLE
Is the following possible with Ory components?

### DISCUSSION THREAD

**Discussion by ldebello-ddl** [ory/kratos#3024] [Q&A] [answered]
Hi all,

I am very newbie in all this IAM world so probably some of my questions are wrong or using not 100% correct terms.I am trying to understand if I can use Ory components for the following:

Context:
I have a webapp which is used from a browser and from api, currently it does not have any kind of authnz.

I want to:
* Offer login with okta before using the webapp from the browser and get an access token to my backend (if I dont understand wrong I can generate this token with Ory components)
* Force my API to receive and authorization header (access token) to authnz and validate this token against Ory
* Offer and endpoint to generate access token for some client apps which are not registered as user in okta.
* In the future offer login with other Identity providers like (github) - (Nice to have)

If someone is able to point me to some examples or specific articles, that would be great or just tell me if I am totally wrong and saying non-sense

Thanks in advance

**Comment by vinckr** [comment, marked as answer]
Hello @ldebello-ddl 
Here are some resources that will help you get started:

- https://www.ory.sh/auth-and-modern-software/
- https://www.ory.sh/docs/security-model
- https://www.ory.sh/zero-trust-api-security-ory-tutorial/
- https://www.ory.sh/docs/getting-started/integrate-auth/go (not sure what is your language of choice)
- https://github.com/ory/examples

Can you share some more details about your application? 
Is it a browser app or mobile/native?
Are you looking to give third parties access to your users data or it is more a "first-party" usecase where your users use their Okta or GitHub accounts to authenticate?

**Reply by ldebello-ddl** [reply]
Thanks for the answer, the articles were very useful.

Our app is a browser app and also we get request from our own CLI and some customer using the API.

The usecase is more like allowing user to use their Okta or GitHub accounts to authenticated and later get an access token (access token handled by us) to use the CLI or the API.

**Reply by vinckr** [reply]
You have the option to get the access/refresh tokens issued by the OIDC provider:
https://www.ory.sh/docs/kratos/social-signin/get-tokens

**Reply by ldebello-ddl** [reply]
If I understand correctly what I got from the social signing is and "ID Token" which later can be introspected against the third party (is that for validation the session?) and then return my own "Access Token" (could I use some of the Ory components to generate this token?) and later authorize the access using the access token (can I use Ory Keto to authorize?)?

**Comment by vinckr** [comment]
Hey @ldebello-ddl 
Ory now supports [OIDC / Social sign-in for native and mobile apps](https://changelog.ory.sh/announcements/social-sign-in-for-native-and-mobile-apps) :tada:
